### PR TITLE
Update script to generate MBdelayed.rda

### PR DESCRIPTION
**Pull request by jdblischak**
I updated `data-raw/DATASET.R` to use the new snakecase function names and lowercase column names. However, I was unable to reproduce the existing object `data/MBdelayed.rda`. My updated version has no events.

Does anyone have any ideas on how to fix the script so that it can reproduce the existing object?

Here are some diagnostics:

```R
# current MBdelayed
summary(MBdelayed)
##       tte               event         Stratum           Treatment        
##  Min.   : 0.09171   Min.   :0.000   Length:200         Length:200        
##  1st Qu.: 5.85919   1st Qu.:0.000   Class :character   Class :character  
##  Median :14.49925   Median :1.000   Mode  :character   Mode  :character  
##  Mean   :16.79899   Mean   :0.705                                        
##  3rd Qu.:27.37013   3rd Qu.:1.000                                        
##  Max.   :35.42239   Max.   :1.000           
str(MBdelayed)
## gropd_df [200 × 4] (S3: grouped_df/tbl_df/tbl/data.frame)
##  $ tte      : num [1:200] 4.68 6.2 35.42 32.74 5.38 ...
##  $ event    : num [1:200] 1 1 1 1 1 1 1 1 1 1 ...
##  $ Stratum  : chr [1:200] "All" "All" "All" "All" ...
##  $ Treatment: chr [1:200] "Experimental" "Control" "Control" "Experimental" ...
##  - attr(*, "groups")= tibble [2 × 3] (S3: tbl_df/tbl/data.frame)
##   ..$ Stratum  : chr [1:2] "All" "All"
##   ..$ Treatment: chr [1:2] "Control" "Experimental"
##   ..$ .rows    :List of 2
##   .. ..$ : int [1:100] 2 3 7 8 9 10 14 15 17 18 ...
##   .. ..$ : int [1:100] 1 4 5 6 11 12 13 16 19 20 ...
##   ..- attr(*, ".drop")= logi TRUE
table(MBdelayed[, c("event", "Stratum", "Treatment")])
## , , Treatment = Control
## 
##      Stratum
## event All
##     0  20
##     1  80
## 
## , , Treatment = Experimental
## 
##      Stratum
## event All
##     0  39
##     1  61

# latest MBdelayed
summary(MBdelayed)
##       tte            event     stratum           treatment        
##  Min.   :24.00   Min.   :0   Length:200         Length:200        
##  1st Qu.:27.11   1st Qu.:0   Class :character   Class :character  
##  Median :29.76   Median :0   Mode  :character   Mode  :character  
##  Mean   :29.88   Mean   :0                                        
##  3rd Qu.:32.71   3rd Qu.:0                                        
##  Max.   :36.30   Max.   :0                   
str(MBdelayed)
## 'data.frame':	200 obs. of  4 variables:
##  $ tte      : num  36.3 36.3 36.1 36.1 36 ...
##  $ event    : num  0 0 0 0 0 0 0 0 0 0 ...
##  $ stratum  : chr  "All" "All" "All" "All" ...
##  $ treatment: chr  "experimental" "control" "control" "experimental" ...
table(MBdelayed[, c("event", "stratum", "treatment")])
## , , treatment = control
## 
##      stratum
## event All
##     0 100
## 
## , , treatment = experimental
## 
##      stratum
## event All
##     0 100
```

### CODE REVIEW

**Comment by nanxstats** [comment]
I can get proper events using an older GitHub release version of simtrial:

```r
remotes::install_github("Merck/simtrial@v0.2.2")
```

then run `data-raw/DATASET.R`. This might help the triage the specific commit(s) that bring the changed behavior. I guess It won't be too difficult to figure out - possibly related to the earlier refactoring effort this Spring.

**Comment by jdblischak** [comment]
Confirmed. I was able to run the code from the tag v0.2.2:

```R
table(MBdelayed[, c("event", "Stratum", "Treatment")])
, , Treatment = Control

     Stratum
event All
    0  23
    1  77

, , Treatment = Experimental

     Stratum
event All
    0  38
    1  62
```

However, the object `data/MBDelayed.rda` was still modified. And note that the numbers don't match the current data set. I suspect that potentially the seed wasn't set the last time it was updated

File: data-raw/DATASET.R [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
-ds <- simPWSurv(
+ds <- sim_pw_surv(
   n = 200,
-  enrollRates = tibble(rate = 200 / 12, duration = 12),
-  failRates = tribble(
-    ~Stratum, ~Period, ~Treatment, ~duration, ~rate,
+  enroll_rate = tibble(rate = 200 / 12, duration = 12),
+  fail_rate = tribble(
+    ~stratum, ~period, ~treatment, ~duration, ~rate,
     "All", 1, "Control", 42, log(2) / 15,
```
Comment (LittleBeannie) [inline data-raw/DATASET.R:10]: Please try update "Control" to "control", and "Experimental" to "experimental".

Comment (jdblischak) [reply data-raw/DATASET.R:10]: Thanks @LittleBeannie! That was indeed the problem. The call to `sim_pw_surv()` used the default value for the argument `block`, which now uses lowercase: `block = c(rep("control", 2), rep("experimental", 2))`

**Comment by jdblischak** [comment]
I also added code to fail early if the input to `sim_pw_surv()` has mismatched treatment names

**Review by nanxstats — APPROVED** [review]
Very nice - great to have this dataset updated and the functions safe guarded.

**Pull request merged**